submission/submission-webapp/src/itest/kotlin/ac/uk/ebi/biostd/itest/test/submission/submit/SubmissionPerformanceTest.kt

```diff
@@ -0,0 +1,73 @@
+package ac.uk.ebi.biostd.itest.test.submission.submit
+
+import ac.uk.ebi.biostd.client.integration.commons.SubmissionFormat
+import ac.uk.ebi.biostd.client.integration.web.BioWebClient
+import ac.uk.ebi.biostd.common.config.FilePersistenceConfig
+import ac.uk.ebi.biostd.itest.common.SecurityTestService
+import ac.uk.ebi.biostd.itest.entities.SuperUser
+import ac.uk.ebi.biostd.itest.itest.ITestListener.Companion.fixedDelayEnv
+import ac.uk.ebi.biostd.itest.itest.ITestListener.Companion.tempFolder
+import ac.uk.ebi.biostd.itest.itest.getWebClient
+import ebi.ac.uk.dsl.tsv.line
+import ebi.ac.uk.dsl.tsv.tsv
+import ebi.ac.uk.io.ext.createFile
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable
+import org.junit.jupiter.api.condition.EnabledIfSystemProperty
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.test.web.server.LocalServerPort
+import org.springframework.context.annotation.Import
+import org.springframework.test.context.junit.jupiter.SpringExtension
+import kotlin.time.ExperimentalTime
+import kotlin.time.measureTime
+
+@OptIn(ExperimentalTime::class)
+@Import(FilePersistenceConfig::class)
+@ExtendWith(SpringExtension::class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class SubmissionPerformanceTest(
+    @Autowired val securityTestService: SecurityTestService,
+    @LocalServerPort val serverPort: Int,
+) {
+    private lateinit var webClient: BioWebClient
+
+    @BeforeAll
+    fun init() {
+        securityTestService.ensureUserRegistration(SuperUser)
+        webClient = getWebClient(serverPort, SuperUser)
+    }
+
+    @Test
+    @EnabledIfEnvironmentVariable(named = fixedDelayEnv, matches = "\\d+")
+    @EnabledIfSystemProperty(named = "enableFire", matches = "true")
+    fun `test with many files`() {
+        val files = 100
+        val delay = System.getenv(fixedDelayEnv).toLong()
+
+        val subFiles = (1..files).map { tempFolder.createFile("$it.txt") }
+        webClient.uploadFiles(subFiles)
+
+        val performanceSubmission = tsv {
+            line("Submission", "SPER-1")
+            line("Title", "Performance Submission")
+            line()
+
+            line("Study")
+            line()
+
+            line("Files")
+            subFiles.forEach { line(it.name) }
```

Comment:
The test looks awesome, I think this is worth measuring. I think it could benefit from adding the files through a file list rather than placing them directly in the submission just to add to the test case the overhead of reading the file list at each stage which is our most common case for big submissions.

Also, I think we could increase the number of files to 1000. It'll add an extra minute but the test will be closer to our actual use case.

WDYT?

Reply:
yes sound good, done.

Reply:
actually just test it with 1000 and runner JVM dies (and I think updating runner to match production machine and get "real performance" is out of the scope)

Reply:
yeah, maybe the execution is too long. I'd still like to see the files in a file list but I'll leave it to you.

Reply:
files are inded in a file list already